iroh-api/src/api.rs

```diff
@@ -201,14 +162,64 @@ impl Api for Iroh {
         .boxed_local()
     }
 
-    fn check(&self) -> BoxFuture<'_, StatusTable> {
+    pub fn check(&self) -> BoxFuture<'_, StatusTable> {
         async { self.client.check().await }.boxed()
     }
 
-    fn watch(
+    pub fn watch(
         &self,
     ) -> LocalBoxFuture<'static, LocalBoxStream<'static, iroh_rpc_client::StatusTable>> {
         let client = self.client.clone();
         async { client.watch().await.boxed_local() }.boxed_local()
     }
+
+    /// High level get, equivalent of CLI `iroh get`
+    pub async fn get<'a>(
+        &self,
+        ipfs_path: &IpfsPath,
+        output_path: Option<&'a Path>,
+    ) -> Result<PathBuf> {
+        if ipfs_path.cid().is_none() {
+            return Err(anyhow!("IPFS path does not refer to a CID"));
+        }
+        let root_path = get_root_path(ipfs_path, output_path);
+        if root_path.exists() {
```

Comment:
Why not use anyhow::ensure https://docs.rs/anyhow/latest/anyhow/macro.ensure.html ?